### PR TITLE
DOC: Additions to month_name & day_name

### LINKED ISSUE #51138: DOC: dt.month_name & dt.day_name locale parameter needs improvment
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
df['DIA_DA_SEMANA'] = df.DAT_EMISSAO.dt.day_name()
df['DIA'] = df.DAT_EMISSAO.dt.day
df['MÊS'] = df.DAT_EMISSAO.dt.month_name(locale='Portuguese') # OUTPUT: Error: unsupported locale setting
df['ANO'] = df.DAT_EMISSAO.dt.year
df
```

### Traceback

<details>

```python
Error                                     Traceback (most recent call last)
Cell In[24], line 3
      1 df['DIA_DA_SEMANA'] = df.DAT_EMISSAO.dt.day_name()
      2 df['DIA'] = df.DAT_EMISSAO.dt.day
----> 3 df['MÊS'] = df.DAT_EMISSAO.dt.month_name(locale='Portuguese')
      4 df['ANO'] = df.DAT_EMISSAO.dt.year
      5 df

File ~/.local/lib/python3.10/site-packages/pandas/core/accessor.py:94, in PandasDelegate._add_delegate_accessors.<locals>._create_delegator_method.<locals>.f(self, *args, **kwargs)
     93 def f(self, *args, **kwargs):
---> 94     return self._delegate_method(name, *args, **kwargs)

File ~/.local/lib/python3.10/site-packages/pandas/core/indexes/accessors.py:126, in Properties._delegate_method(self, name, *args, **kwargs)
    123 values = self._get_values()
    125 method = getattr(values, name)
--> 126 result = method(*args, **kwargs)
    128 if not is_list_like(result):
    129     return result

File ~/.local/lib/python3.10/site-packages/pandas/core/indexes/extension.py:98, in _inherit_from_data.<locals>.method(self, *args, **kwargs)
     96 if "inplace" in kwargs:
     97     raise ValueError(f"cannot use inplace with {type(self).__name__}")
---> 98 result = attr(self._data, *args, **kwargs)
     99 if wrap:
    100     if isinstance(result, type(self._data)):

File ~/.local/lib/python3.10/site-packages/pandas/core/arrays/datetimes.py:1212, in DatetimeArray.month_name(self, locale)
   1175 """
   1176 Return the month names with specified locale.
   1177 
   (...)
   1208 Index(['January', 'February', 'March'], dtype='object')
   1209 """
   1210 values = self._local_timestamps()
-> 1212 result = fields.get_date_name_field(
   1213     values, "month_name", locale=locale, reso=self._reso
   1214 )
   1215 result = self._maybe_mask_results(result, fill_value=None)
   1216 return result

File ~/.local/lib/python3.10/site-packages/pandas/_libs/tslibs/fields.pyx:179, in pandas._libs.tslibs.fields.get_date_name_field()

File ~/.local/lib/python3.10/site-packages/pandas/_libs/tslibs/fields.pyx:622, in pandas._libs.tslibs.fields._get_locale_names()
...
    618     # convert to string
    619     locale = normalize(_build_localename(locale))
--> 620 return _setlocale(category, locale)

Error: unsupported locale setting
```

</details>

### Issue Description

I'm trying to set a month column for my DF with portuguese names by doing as the [docs](https://pandas.pydata.org/docs/reference/api/pandas.Series.dt.month_name.html) said, but it didn't work, I even tried with `'English'` which says it is the default but it doesn't work either.

### Expected Behavior

I expect it to return a serialized column with month names in Portuguese

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 2e218d10984e9919f0296931d92ea851c6a6faf5
python           : 3.10.9.final.0
python-bits      : 64
OS               : Linux
OS-release       : 6.0.12-76060006-generic
Version          : #202212290932~1674139725~22.04~ca93ccf SMP PREEMPT_DYNAMIC Thu J
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : pt_BR.UTF-8

pandas           : 1.5.3
numpy            : 1.23.5
pytz             : 2022.7
dateutil         : 2.8.2
setuptools       : 65.6.3
pip              : 22.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.9.1
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.7.0
pandas_datareader: None
bs4              : 4.11.1
bottleneck       : 1.3.5
brotli           : 
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : 3.6.2
numba            : None
numexpr          : 2.8.4
...
xlrd             : None
xlwt             : None
zstandard        : None
tzdata           : 2022.7
[/home/eduardochaves/miniconda3/envs/datasets_procenge/lib/python3.10/site-packages/_distutils_hack/__init__.py:33](https://file+.vscode-resource.vscode-cdn.net/home/eduardochaves/miniconda3/envs/datasets_procenge/lib/python3.10/site-packages/_distutils_hack/__init__.py:33): UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

</details>

### PR DESCRIPTION
**Pull request by eduardochaves1**
- [x] closes #51138
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

In the issue #51138 I was trying to set a column in my DF for the month and day's names in portuguese, then a good guy told me about the `locale -a` command from terminal to look for the right way to set the `locale` string parameter for my code to work properly.

So I thought I would be nice if the respective docs, [month_name](https://pandas.pydata.org/docs/reference/api/pandas.Series.dt.month_name.html) & [day_name](https://pandas.pydata.org/docs/reference/api/pandas.Series.dt.day_name.html), had these informations for the user to know about.

So I added these infos plus a quick example for each method in the docstrings from `/pandas/core/arrays/datetimes.py`

**PS.:** I just don't know why the `validade_docstrings.py` don't recognize the changes I made in the method's docstring...

<details>
<summary>Output: <code>validate_docstrings.py pandas.Series.dt.month_name</code></summary>

Screenshot from 2023-02-03 21-02-04

</details>

<details>
<summary>Output: <code>validate_docstrings.py pandas.Series.dt.day_name</code></summary>

Screenshot from 2023-02-03 21-00-58

</details>

<details>
<summary>Output: <code>pre-commit run --files pandas/core/arrays/datetimes.py</code></summary>

Screenshot from 2023-02-03 22-11-09

</details>

### CODE REVIEW

**Comment by eduardochaves1** [comment]
The CI is getting a `UNEXPECTED EXCEPTION: Error('unsupported locale setting')` error for my example with `locale='pt_BR.utf8'`. And I don't know exactly why, cause I tried the exact same code on my machine and it worked. Maybe this locale isn't installed on the machine in which does the CI.

**Comment by mroeschke** [comment]
You will probably need to add `   # doctest: +SKIP` to the example due to the locale being unavailable

**Comment by eduardochaves1** [comment]
> You will probably need to add ` # doctest: +SKIP` to the example due to the locale being unavailable

I added it in the same line where the function was called with `locate='pt_BR.utf8` (for both month_name & day_name) and the CI went well.

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -1202,7 +1202,8 @@ def month_name(self, locale=None) -> npt.NDArray[np.object_]:
         ----------
         locale : str, optional
             Locale determining the language in which to return the month name.
-            Default is English locale.
+            Default is English locale (``'en_US.utf8'``). Use the command
+            ``locale -a`` on your terminal to look for your locale language code.
```
Comment (mroeschke) [inline pandas/core/arrays/datetimes.py:1206]:
```suggestion
            ``locale -a`` on your terminal on Unix systems to find your locale language code.
```

Could you change the other usages of this below too?

Comment (eduardochaves1) [reply pandas/core/arrays/datetimes.py:1206]: Even better! I'll change that on the day_name docstring as well.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @eduardochaves1

**Pull request merged**